lighthouse-core/test/runner.js

```diff
@@ -77,67 +67,114 @@ describe('Runner', () => {
     });
   });
 
-  it('throws for unknown gatherers', () => {
+  it('throws when given neither passes nor artifacts', () => {
     const url = 'https://example.com';
     const flags = {
       auditWhitelist: null
     };
-    const config = {
-      passes: [{
-        gatherers: ['fuzz']
-      }],
+    const config = new Config({
       audits: [
         'is-on-https'
       ]
-    };
+    }, flags.auditWhitelist);
 
     return assert.throws(_ => Runner.run(fakeDriver, {url, config, flags}),
-        /Unable to locate/);
+        /The config must provide passes/);
   });
 
-  it('throws when given neither passes nor artifacts', () => {
+  it('accepts existing artifacts', () => {
     const url = 'https://example.com';
     const flags = {
       auditWhitelist: null
     };
-    const config = {
+    const config = new Config({
       audits: [
         'is-on-https'
-      ]
+      ],
+
+      artifacts: {
+        HTTPS: true
+      }
+    }, flags.auditWhitelist);
+
+    return assert.doesNotThrow(_ => Runner.run(fakeDriver, {url, config, flags}));
+  });
+
+  it('accepts trace artifacts as paths and outputs appropriate data', () => {
+    const url = 'https://example.com';
+    const flags = {
+      auditWhitelist: null
     };
 
-    return assert.throws(_ => Runner.run(fakeDriver, {url, config, flags}),
-        /The config must provide passes/);
+    const config = new Config({
+      audits: [
+        'user-timings'
+      ],
+
+      artifacts: {
+        traceContents: path.join(__dirname,
+                           '/fixtures/traces/trace-user-timings.json')
+      }
+    }, flags.auditWhitelist);
+
+    return Runner.run(fakeDriver, {url, config, flags}).then(results => {
+      assert.equal(results[0].value, 2);
+      assert.equal(results[0].name, 'user-timings');
+    });
   });
 
-  it('accepts existing artifacts', () => {
+  it('fails gracefully when trace artifacts are not provided but artifacts are present', () => {
```

Comment:
can you clarify this description for what's being tested? `'empty artifacts object is present'`, maybe?